File: sdk/java/src/main/java/com/sgraal/SgraalClient.java
package com.sgraal;
import java.net.http.*;
import java.net.URI;
public class SgraalClient {
    private final String apiKey;
    private final String baseUrl;
    public SgraalClient(String apiKey) { this(apiKey, "https://api.sgraal.com"); }
    public SgraalClient(String apiKey, String baseUrl) { this.apiKey = apiKey; this.baseUrl = baseUrl; }
    public String preflight(String jsonBody) throws Exception {
        var client = HttpClient.newHttpClient();
        var request = HttpRequest.newBuilder().uri(URI.create(baseUrl + "/v1/preflight"))
            .header("Authorization", "Bearer " + apiKey).header("Content-Type", "application/json")
            .POST(HttpRequest.BodyPublishers.ofString(jsonBody)).build();
        return client.send(request, HttpResponse.BodyHandlers.ofString()).body();
    }
    public String heal() { throw new UnsupportedOperationException("Coming in next release"); }
    public String batch() { throw new UnsupportedOperationException("Coming in next release"); }
}

File: sdk/dotnet/Sgraal/SgraalClient.cs
namespace Sgraal;
using System.Net.Http;
using System.Net.Http.Json;
using System.Text.Json;
using Sgraal.Models;

public class SgraalClient
{
    private readonly HttpClient _http;
    private readonly string _baseUrl;

    public SgraalClient(string apiKey, string baseUrl = "https://api.sgraal.com")
    {
        _baseUrl = baseUrl;
        _http = new HttpClient();
        _http.DefaultRequestHeaders.Add("Authorization", $"Bearer {apiKey}");
    }

    public async Task<PreflightResponse?> PreflightAsync(PreflightRequest request)
    {
        var resp = await _http.PostAsJsonAsync($"{_baseUrl}/v1/preflight", request);
        resp.EnsureSuccessStatusCode();
        return await resp.Content.ReadFromJsonAsync<PreflightResponse>();
    }
}
